lib/Conversion/ExportVerilog/ExportVerilog.cpp

```diff
@@ -2626,13 +2627,21 @@ void StmtEmitter::emitStatementExpression(Operation *op) {
     if (emitDeclarationForTemporary(op))
       return;
     os << " = ";
+    if (auto add = dyn_cast<AddOp>(op))
```

Comment:
IT would be better to not be probing op type here.  I do see that this kind of plumbing is partially done for isSelfDeterminedUnsignedValue in emitSubExpr.

ExprEmitter::emitBinary might be a better place to handle this.  This function already has to handle a similar issue with signedness.

Reply:
I don't think `18'(a * b )` is a widely accepted construct in Verilog.